docs/manual/core-concepts/getters-setters-virtuals.md

```diff
@@ -0,0 +1,161 @@
+
+## Combining getters and setters
+
+Getters and setters can be both defined in the same field.
+
+For the sake of an example, let's say we are modeling a `Post`, whose `content` is a text of unlimited length. To improve memory usage, let's say we want to store a gzipped version of the content.
+
+*Note: modern databases should do some compression automatically in these cases. Please note that this is just for the sake of an example.*
+
+```js
+const { gzipSync, gunzipSync } = require('zlib');
+
+const Post = sequelize.define('post', {
+  content: {
+    type: DataTypes.TEXT,
+    get() {
+      const storedValue = this.getDataValue('content');
+      const gzippedBuffer = Buffer.from(storedValue, 'base64');
+      const unzippedBuffer = gunzipSync(gzippedBuffer);
+      return unzippedBuffer.toString();
+    },
+    set(value) {
+      const gzippedBuffer = gzipSync(value);
+      this.setDataValue('content', gzippedBuffer.toString('base64'));
+    }
+  }
+});
+```
+
+With the above setup, whenever we try to interact with the `content` field of our `Post` model, Sequelize will automatically handle the custom getter and setter. For example:
+
+```js
+const post = await Post.create({ content: 'Hello everyone!' });
+
+console.log(post.content); // 'Hello everyone!'
+// Everything is happening under the hood, so we can even forget that the
+// content is actually being stored as a gzipped base64 string!
+
+// However, if we are really curious, we can get the 'raw' data...
+console.log(post.getDataValue('content'));
+// Output: 'H4sIAAAAAAAACvNIzcnJV0gtSy2qzM9LVQQAUuk9jQ8AAAA='
+```
+
+## Virtual fields
+
+Virtual fields are fields that Sequelize populates under the hood, but in reality they don't even exist in the database.
+
+For example, let's say we have the `firstName` and `lastName` attributes for a User.
+
+*Again, this is [only for the sake of an example](https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/).*
+
+It would be nice to have a simple way to obtain the *full name* directly! We can combine the idea of `getters` with the special data type Sequelize provides for this kind of situation: `DataTypes.VIRTUAL`:
+
+```js
+const { DataTypes } = require("sequelize");
+
+const User = sequelize.define('user', {
+  firstName: DataTypes.TEXT,
+  lastName: DataTypes.TEXT,
+  fullName: {
+    type: DataTypes.VIRTUAL,
+    get() {
+      return `${this.firstName} ${this.lastName}`;
+    },
+    set(value) {
+      throw new Error('Do not try to set the `fullName` value!');
+    }
+  }
+});
+```
+
+The `VIRTUAL` field does not cause a column in the table to exist. In other words, the model above will not have a `fullName` column. However, it will appear to have it!
+
+```js
+const user = await User.create({ firstName: 'John', lastName: 'Doe' });
+console.log(user.fullName); // 'John Doe'
+```
+
+## Deprecated: `getterMethods` and `setterMethods`

```

Comment:
Not deprecated at code level, right?

Reply:
Indeed... Not deprecated (yet?). I will rewrite this part a bit, and hopefully [deprecate them](https://github.com/orgs/sequelize/teams/docs/discussions/4) in the near future.

Reply:
[Fixed](https://github.com/sequelize/sequelize/pull/11825/commits/7f6eba41c88894982763b41bbbfc4579e8f8a04e)